Sources/UI/TruncatingLabel/TruncatingLabel.swift

```diff
@@ -0,0 +1,325 @@
+// there is enough space for both of them in two lines.
+//  ---------------------------------------------
+// |   This is a longer single-line text.        |
+// |                           secondary label   |
+//  ---------------------------------------------
+//
+//  ---------------------------------------------
+// |   This is a longer two-line text, lorem     |
+// |   ipsum dolor.            secondary label   |
+//  ---------------------------------------------
+//
+// 3. Strings do not fit into a single line and there is also
+// not enough space for the primary string in the second line.
+// Primary string, therefore, has to be truncated with '...'.
+//  ---------------------------------------------
+// |   This is a long text that has to be split  |
+// |   into two lines, ho...    secondary label  |
+//  ---------------------------------------------
+//
+// 4. There is not enough space for the secondary text
+// in the whole line. Truncate primary text in the first line.
+//  ---------------------------------------------
+// |   This is a long text that has to be sp...  |
+// |   secondary label is too long to be disp... |
+//  ---------------------------------------------
+//
+public class TruncatingLabel: UIView {
+  // @NOTE: - For autolayout to work properly, at least the width of the view
+  // should be 'statically' determined (that is, layoutSubviews() should
+  // receive a frame with valid width) by the constraint setup, i.e., bind
+  // the leading and trailing constraints to the parent view. The height of the
+  // view is dynamic and the intrinsic content size updated accordingly.
+  
+  private var primaryTextAttributes: [NSAttributedString.Key: Any] = [
+    .font: UIFont.systemFont(ofSize: 14),
+    .foregroundColor: UIColor.gray
+  ]
+  private var secondaryTextAttributes: [NSAttributedString.Key: Any] = [
+    .font: UIFont.systemFont(ofSize: 12),
+    .foregroundColor: UIColor.blue
+  ]
+  
+  public var primaryColor: UIColor {
+    get {
+      primaryTextAttributes[.foregroundColor]! as! UIColor
+    }
+    set {
+      primaryTextAttributes[.foregroundColor] = newValue
+      setNeedsDisplay()
+    }
+  }
+  public var primaryFont: UIFont {
+    get {
+      primaryTextAttributes[.font]! as! UIFont
+    }
+    set {
+      primaryTextAttributes[.font] = newValue
+      internalIntrinsicContentSize = calculateIntrinsicContentSize()
+      setNeedsDisplay()
+    }
+  }
+  public var secondaryColor: UIColor {
+    get {
+      secondaryTextAttributes[.foregroundColor]! as! UIColor
+    }
+    set {
+      secondaryTextAttributes[.foregroundColor] = newValue
+      setNeedsDisplay()
+    }
+  }
+  public var secondaryFont: UIFont {
+    get {
+      secondaryTextAttributes[.font]! as! UIFont
+    }
+    set {
+      secondaryTextAttributes[.font] = newValue
+      internalIntrinsicContentSize = calculateIntrinsicContentSize()
+      setNeedsDisplay()
+    }
+  }
+  public var gravity: Gravity = .left {
```

Comment:
I would add a getter method so the receiver knows exactly what is gravity applied to 🤷‍♂️ 
```swift
/// Gravity is a rule that specifies the sticking point for the secondary text in relation to the primary text. The default value is `left`.
public var gravity: Gravity = .left {
```